perception/autoware_traffic_light_arbiter/src/signal_match_validator.cpp

```diff
@@ -155,10 +155,11 @@ std::unordered_set<lanelet::Id> create_signal_id_set(
   return signal_id_set;
 }
 
-// Returns the signal with the highest confidence elements, considering a external priority
+// Returns the signal with the highest confidence elements, considering source priority
 TrafficSignal get_highest_confidence_signal(
   const std::optional<TrafficSignal> & perception_signal,
-  const std::optional<TrafficSignal> & external_signal, const bool external_priority)
+  const std::optional<TrafficSignal> & external_signal,
+  const autoware::traffic_light::SourcePriority source_priority)
 {
   // Returns the existing signal if only one of them exists
```

Comment:
[imho]
Though I know there is escape before the usage of this function , I personally want protection or assertion to prevent case which both signals are nullopt. `if (!perception_signal && !external_signal) ...`

Reply:
Thank you! This makes sense. However, I think this should be done in another PR. Assertion won't make much difference in release build and proper protection will require using optional in the return type or something of that kind.